### PR TITLE
Add TextExtract Kotlin sample

### PR DESCRIPTION
**Pull request by datalogics-saharay**
Let's add a Kotlin sample that shows how to extract text from tagged and untagged PDF documents using the Java Maven APDFL package!

**No Jira issue**

### CODE REVIEW

**Review by datalogics-robb — APPROVED** [review]

File: TextExtract/src/main/kotlin/com/datalogics/pdfl/samples/TextExtract.kt [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+        val wordFinder = WordFinder(doc, WordFinderVersion.LATEST, wordConfig)
+
+        if (docIsTagged)
+            extractTextTagged(doc, wordFinder)
+        else
+            extractTextUntagged(doc, wordFinder)
+
+        doc.close()
+    }finally {
```
Comment (datalogics-robl) [inline TextExtract/src/main/kotlin/com/datalogics/pdfl/samples/TextExtract.kt]: Is there a coding standard for Kotlin that covers spacing, indentation, etc.? The uneven spacing around the `finally` keyword really stands out here.

More generally, samples should have on their best formal wear, so we should really get the details right: for example, note the space after `if` on line 77 versus the lack thereof on lines 49-50-51.

Really, I'm advocating for finding a Kotlin linter/formatter and running this sample through it.

File: TextExtract/src/main/kotlin/com/datalogics/pdfl/samples/TextExtract.kt [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+import com.datalogics.PDFL.*
+import java.io.FileOutputStream
+import java.io.OutputStreamWriter
+
+/*
+*
+* This program pulls text from a PDF file and exports it to a text file (TXT).
+* It will open a PDF file called Constitution.PDF and create an output file called
+* TextExtract-untagged-out.txt. The export file includes page number references, and
+* the text is produced using standard Times Roman encoding. The program is also written
```
Comment (datalogics-robl) [inline TextExtract/src/main/kotlin/com/datalogics/pdfl/samples/TextExtract.kt]: I see we copied this from another sample, but I question if "Times Roman" is the right term. "Times Roman" is a font; "Roman" is an encoding, more or less a synonym for "Latin", and is probably what is meant.

**Review by datalogics-robl — CHANGES_REQUESTED** [review]
My only suggestions are formatting issues and possibly a wrong name in the comments.

File: TextExtract/src/main/kotlin/com/datalogics/pdfl/samples/TextExtract.kt (hunk 1 above)
Comment (datalogics-saharay) [reply TextExtract/src/main/kotlin/com/datalogics/pdfl/samples/TextExtract.kt]: Good suggestion, done! Luckily, IntelliJ had a built-in Kotlin formatter: https://discuss.kotlinlang.org/t/kotlin-code-auto-format/407.

File: TextExtract/src/main/kotlin/com/datalogics/pdfl/samples/TextExtract.kt (hunk 2 above)
Comment (datalogics-saharay) [reply TextExtract/src/main/kotlin/com/datalogics/pdfl/samples/TextExtract.kt]: Fixed

**Review by datalogics-robl — APPROVED** [review]

**Pull request merged**